server/service/src/stocktake_line/validate.rs

```diff
@@ -3,15 +3,17 @@ use repository::{
         InventoryAdjustmentReason, InventoryAdjustmentReasonFilter,
         InventoryAdjustmentReasonRepository,
     },
-    EqualFilter, InventoryAdjustmentReasonType, RepositoryError, StockLineRow, StocktakeLineRow,
-    StocktakeLineRowRepository, StorageConnection,
+    EqualFilter, InventoryAdjustmentReasonType, RepositoryError, StockLineRow, StocktakeLine,
+    StocktakeLineFilter, StocktakeLineRepository, StocktakeLineRow, StorageConnection,
 };
 
 pub fn check_stocktake_line_exist(
     connection: &StorageConnection,
     id: &str,
-) -> Result<Option<StocktakeLineRow>, RepositoryError> {
-    StocktakeLineRowRepository::new(&connection).find_one_by_id(id)
+) -> Result<Option<StocktakeLine>, RepositoryError> {
```

Comment:
All these changes are making the repo layer make way more sense to me, I was confused why we had domain structs but seemingly frequently didn't use them, just using the repo row structs instead. In fairness I think in many cases by using the repo we're making a bunch of joins that may not be necessary, including this case.